packages/replay/src/coreHandlers/handleXhr.ts

```diff
@@ -1,76 +1,48 @@
+import type { HandlerDataXhr } from '@sentry/types';
+
 import type { ReplayContainer, ReplayPerformanceEntry } from '../types';
 import { createPerformanceSpans } from '../util/createPerformanceSpans';
 import { shouldFilterRequest } from '../util/shouldFilterRequest';
 
-// From sentry-javascript
-// e.g. https://github.com/getsentry/sentry-javascript/blob/c7fc025bf9fa8c073fdb56351808ce53909fbe45/packages/utils/src/instrument.ts#L180
-type XHRSendInput = null | Blob | BufferSource | FormData | URLSearchParams | string;
-
-interface SentryWrappedXMLHttpRequest extends XMLHttpRequest {
-  [key: string]: unknown;
-  __sentry_xhr__?: {
-    method?: string;
-    url?: string;
-    status_code?: number;
-    body?: XHRSendInput;
-    startTimestamp?: number; // This is unique to replay SDK
-  };
-  // If Sentry key appears in request, don't capture as request
-  // See https://github.com/getsentry/sentry-javascript/blob/c7fc025bf9fa8c073fdb56351808ce53909fbe45/packages/utils/src/instrument.ts#L236
-  __sentry_own_request__?: boolean;
-}
-
-interface XhrHandlerData {
-  args: [string, string];
-  xhr: SentryWrappedXMLHttpRequest;
-  startTimestamp: number;
-  endTimestamp?: number;
-}
-
-function handleXhr(handlerData: XhrHandlerData): ReplayPerformanceEntry | null {
-  if (handlerData.xhr.__sentry_own_request__) {
-    // Taken from sentry-javascript
-    // Only capture non-sentry requests
-    return null;
-  }
-
-  if (handlerData.startTimestamp) {
-    // TODO: See if this is still needed
-    handlerData.xhr.__sentry_xhr__ = handlerData.xhr.__sentry_xhr__ || {};
-    handlerData.xhr.__sentry_xhr__.startTimestamp = handlerData.startTimestamp;
-  }
+/** only exported for tests */
+export function handleXhr(handlerData: HandlerDataXhr): ReplayPerformanceEntry | null {
+  const { startTimestamp, endTimestamp, xhr } = handlerData;
 
-  if (!handlerData.endTimestamp) {
+  if (!startTimestamp || !endTimestamp || !xhr.__sentry_xhr__) {
```

Comment:
Not 100% sure, but I don't think the "moving timestamps around & checking differnent places" stuff is really needed here (anymore), this should always be set, and IMHO if one of these fails we just bail out.